pkgs/community/swarmauri_community/vector_stores/DuckDBVectoreStore.py

```diff
@@ -0,0 +1,331 @@
+import json
+import duckdb
+import numpy as np
+import os
+from typing import List, Union, Literal, Dict, Any, Optional
+from pydantic import BaseModel, Field
+
+from swarmauri.documents.concrete.Document import Document
+from swarmauri.embeddings.concrete.Doc2VecEmbedding import Doc2VecEmbedding
+from swarmauri.distances.concrete.CosineDistance import CosineDistance
+from swarmauri.vectors.concrete.Vector import Vector
+
+from swarmauri.vector_stores.base.VectorStoreBase import VectorStoreBase
+from swarmauri.vector_stores.base.VectorStoreRetrieveMixin import (
+    VectorStoreRetrieveMixin,
+)
+from swarmauri.vector_stores.base.VectorStoreSaveLoadMixin import (
+    VectorStoreSaveLoadMixin,
+)
+
+
+class DuckDBVectorStore(
+    VectorStoreSaveLoadMixin, VectorStoreRetrieveMixin, VectorStoreBase, BaseModel
+):
+    type: Literal["DuckDBVectorStore"] = "DuckDBVectorStore"
+    db_path: str
+    vector_size: int
+    metric: str
+    client: Optional[duckdb.DuckDBPyConnection] = Field(default=None, exclude=True)
+
+    class Config:
+        arbitrary_types_allowed = True
+
+    def __init__(
+        self, db_path: str, vector_size: int, metric: str = "cosine", **kwargs
+    ):
+        super().__init__(
+            db_path=db_path, vector_size=vector_size, metric=metric, **kwargs
```

Comment:
If its defined at class then we dont need to define here

Reply:
That's true
Thanks for that.
Changes already done